src/flexpath.cpp

```diff
@@ -822,6 +841,13 @@ ErrorCode FlexPath::to_gds(FILE* out, double scaling) {
                                    end_type,
                                    8,
                                    0x0F03};
+
+        uint16_t sname_start[] = {(uint16_t)(4 + len), 0x1206};
+        uint16_t pxxdata_start[] = {4, 0x6206};
+
+        big_endian_swap16(sname_start, COUNT(sname_start));
+        big_endian_swap16(pxxdata_start, COUNT(pxxdata_start));
```

Comment:
This is strange to me… PXXDATA has members of several different sizes. Why are we swapping them considering all as 16-bit long?

Reply:
Since you did this for `buffer_start`, i assumed it was the right thing to do in this case

Reply:
I see. In that case, you probably want to swap each member with its specific size function, e.g. doubles with use swap64, etc.

Reply:
I ended up not swapping the PXXData at all, is this still an issue?

Reply:
Yes, because it won't work for files saved in big endian architectures.

Reply:
I am a bit out of my depth here, could you point me in the right direction to fix this?

Reply:
I have removed some unused headers and the unused while loop. When I swap the `serialized_pxxdata` or swap each member of PXXData, I get an incorrect output. I'm not sure what else to try.

Reply:
Probably something like this (I have not tested it):

```cpp
        // No need to memcpy
        PXXDATA pxxdata = convert_to_pxxdata(raith_data);
        uint8_t* serialized_pxxdata = (uint8_t*)&pxxdata;

        // ...

        big_endian_swap16(sname_start, COUNT(sname_start));
        // PXXData.unused
        big_endian_swap16((uint16_t*)(serialized_data + offsetof(PXXData, unused)), 1);
        // PXXData.pitch_*
        big_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, pitch_parallel_to_path)), 3);
        // PXXData.periods .. ret_pixel_count
        big_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, periods)), 5);
        // PXXData.ret_stage_speed, ret_dwell_time
        big_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, ret_stage_speed)), 2);
        // PXXData.revision
        big_endian_swap16((uint16_t*)(serialized_data + offsetof(PXXData, revision)), 1);
```

Reply:
As per the requirements of the Raith software, the data should be encoded with little endian. This code results in the correct output for me, but I could see this being an issue having little endian and big endian in the same file? When I use big endian here instead of little endian, the Raith software doesn't load the GDS file, so this doesn't work. 
```cpp
little_endian_swap16((uint16_t*)(serialized_data + offsetof(PXXData, unused)), 1);

little_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, pitch_parallel_to_path)), 1);
little_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, pitch_perpendicular_to_path)), 1);
little_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, pitch_scale)), 1);

little_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, periods)), 1);
little_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, grating_type)), 1);
little_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, dots_per_cycle)), 1);
little_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, ret_base_pixel_count)), 1);
little_endian_swap32((uint32_t*)(serialized_data + offsetof(PXXData, ret_pixel_count)), 1);

little_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, ret_stage_speed)), 1);
little_endian_swap64((uint64_t*)(serialized_data + offsetof(PXXData, ret_dwell_time)), 1);

little_endian_swap16((uint16_t*)(serialized_data + offsetof(PXXData, revision)), 1);
}
```
I'm not sure what would work here, does all the data need to be swapped in big endian?

Reply:
If Raith specifies little indian, then your changes are correct!

Reply:
Thats me pushed my changes now